### PR TITLE
fix: IAM Roles  Permission Boundary

### PR DESCRIPTION
**Pull request by bbernays**
#### Summary

This PR works around a known issue where `ListRoles` doesn't return all of the necessary information and the only way of getting the Boundary Permission information is by doing a `GetRole` for each resource.

This will mean that there will be A LOT more calls (one more for each IAM role)

---

Use the following steps to ensure your PR is ready to be reviewed

- [x] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [x] Test locally on your own infrastructure
- [x] Run `go fmt` to format your code 🖊
- [x] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation)) 
- [x] Update or add tests 🧪
- [x] Ensure the status checks below are successful ✅

### CODE REVIEW

File: plugins/source/aws/docs/tables/aws_iam_users.md [hunk 1]
```diff
@@ -4,27 +4,25 @@
...
+|password_next_rotation|timestamp without time zone||
+|mfa_active|boolean||
+|access_key1_active|boolean||
+|access_key2_active|boolean||
+|access_key1_last_rotated|timestamp without time zone||
+|access_key2_last_rotated|timestamp without time zone||
+|cert1_active|boolean||
+|cert2_active|boolean||
+|cert1_last_rotated|timestamp without time zone||
+|cert2_last_rotated|timestamp without time zone||
```
Comment (bbernays) [inline plugins/source/aws/docs/tables/aws_iam_users.md:28]: Add Descriptions to gen.hcl

File: plugins/source/aws/resources/services/iam/user_policies.go [hunk 2]
```diff
@@ -1,109 +0,0 @@
...
-				Description: "The AWS Account ID of the resource.",
-				Type:        schema.TypeString,
-				Resolver:    client.ResolveAWSAccount,
-			},
-			{
-				Name:        "user_id",
-				Description: "user ID the policy belongs too.",
-				Type:        schema.TypeString,
-				Resolver:    schema.ParentResourceFieldResolver("id"),
-			},
```
Comment (bbernays) [inline plugins/source/aws/resources/services/iam/user_policies.go:40]: These fields have been removed as they are relations and can be grabbed from the parent table

File: plugins/source/aws/client/helpers.go [hunk 3]
```diff
@@ -59,6 +59,7 @@ const (
 	SESService                  AWSService = "ses"
 	WAFRegional                 AWSService = "waf-regional"
 	WorkspacesService           AWSService = "workspaces"
+	IamService                  AWSService = "iam"
```
Comment (disq) [inline plugins/source/aws/client/helpers.go:62]: I would keep this list sorted (move this line up after GuardDuty)

Comment (bbernays) [reply plugins/source/aws/client/helpers.go]: fixed

**Review by yevgenypats — APPROVED** [review]

**Pull request closed without merging**